File: frontend/src/types/mission.ts
import type { GeoPoint, Iso8601, Uuid } from './common';

export type MissionStatus = 'planned' | 'active' | 'completed' | 'aborted';

export interface AreaOfOperations {
  name: string;
  polygon: GeoPoint[];
}

export interface Mission {
  id: Uuid;
  name: string;
  status: MissionStatus;
  area: AreaOfOperations;
  started_at: Iso8601;
  ended_at: Iso8601 | null;
  sensor_ids: Uuid[];
  operator_ids: Uuid[];
}

File: backend/app/domain/models/mission.py
"""Mission: an operational context grouping sensors, areas and personnel."""
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from uuid import UUID

from app.domain.enums import MissionStatus
from app.domain.models.detection import GeoPoint


@dataclass(frozen=True, slots=True)
class AreaOfOperations:
    """Polygon defining the geographic scope of a mission."""

    name: str
    polygon: tuple[GeoPoint, ...]


@dataclass(frozen=True, slots=True)
class Mission:
    """A bounded operation. All detections are scoped to a mission for audit."""

    id: UUID
    name: str
    status: MissionStatus
    area: AreaOfOperations
    started_at: datetime
    ended_at: datetime | None
    sensor_ids: tuple[UUID, ...] = field(default_factory=tuple)
    operator_ids: tuple[UUID, ...] = field(default_factory=tuple)
